File: projects/lostengine/lost/lua/lost/io/rg/Clear.lua
module("lost.io", package.seeall)

require("lost.io.Loader")

-- calls the glClear command with the given mask
function Loader:Clear(definition)
  local mask = definition["mask"]
  if mask == nil then
    error("mask is required for rg:Clear")
  end
  local result = lost.rg.Clear.create(definition["mask"])
  
  local active = definition["active"]
  local name = definition["name"]
  if active ~= nil then
    result.active = active
  end
  if name ~= nil then
    result.name = name
  end
  return result
end

File: projects/lostengine/lost/rg/Clear.cpp
#include "lost/rg/Clear.h"
#include "lost/gl/Context.h"

namespace lost
{
namespace rg
{

Clear::Clear(GLbitfield inMask)
{
  name = "clear";
  mask = inMask;
}

NodePtr Clear::create(GLbitfield inMask)
{
  return ClearPtr(new Clear(inMask));
}

void Clear::process(gl::ContextPtr ctx)
{
  if(!active) return;
  ctx->clear(mask);
}

}
}